glib/gsocket.go

```diff
@@ -0,0 +1,42 @@
+package glib
+
+// #include <gio/gio.h>
+import "C"
+import (
+	"errors"
+	"syscall"
+	"unsafe"
+)
+
+type Socket struct {
+	ptr *Object
```

Comment:
Embed the object if it is the base class in C. That way all Object methods are also exposed for the user.

```suggestion
	*Object
```

Reply:
done

Reply:
Not exactly, you embedded `Object` instead of `*Object`. This will cause finalizers to run early because you copy the struct via `*Take`, which will result in segfaults.